src/passes/OptimizeInstructions.cpp

```diff
@@ -570,6 +570,22 @@ struct OptimizeInstructions
         return ret;
       }
     } else if (auto* unary = curr->dynCast<Unary>()) {
+      if (unary->op == EqZInt32 || unary->op == EqZInt64) {
+        using namespace Match;
+        Const* c;
+        Binary* inner;
+        Expression* left;
+        // eqz((signed)x % C_pot)   ==>   eqz(x & (C_pot - 1))
+        if (matches(unary->value,
+                    Match::binary(
+                      &inner, Abstract::RemS, any(&left), constant(&c)))) {
```

Comment:
You could use `ival` instead of `constant` to make this more compact.

Reply:
It will be impossible after [this](https://github.com/WebAssembly/binaryen/pull/3153#discussion_r494742192) I guess